## Ejercicio 1

### Con el archivo Summer-Olympic-medals-1976-to-2008, construir un objeto de series de tiempo con el índice igual al año de los juegos olímpicos.

In [197]:
import pandas as pd
import numpy as np


In [24]:
data_set= pd.read_csv('Summer-Olympics.csv',encoding='cp1252')

In [25]:
data_set.head(3)

City    Year     Sport Discipline           Event  \
0  Montreal  1976.0  Aquatics     Diving  3m springboard   
1  Montreal  1976.0  Aquatics     Diving  3m springboard   
2  Montreal  1976.0  Aquatics     Diving  3m springboard   

                Athlete Gender Country_Code        Country Event_gender  \
0       KÖHLER, Christa  Women          GDR   East Germany            W   
1   KOSENKOV, Aleksandr    Men          URS   Soviet Union            M   
2  BOGGS, Philip George    Men          USA  United States            M   

    Medal  
0  Silver  
1  Bronze  
2    Gold

In [26]:
data_set=data_set.set_index(['Year'])

# Ejercicio 2

## Separar nombres de apellidos en dos columnas distintas usando la coma como separador

In [234]:
data_set=data_set.rename(columns={'Athlete':'First_Name'})

In [235]:
data_set[['Last_Name','First_Name']]=data_set['First_Name'].str.split(',',1,expand=True)

In [236]:
columns_name=data_set.columns

In [237]:
data_set=data_set.reindex(columns=['City', 'Sport', 'Discipline', 'Event', 'First_Name', 'Last_Name', 'Gender',
       'Country_Code', 'Country', 'Event_gender', 'Medal'])

In [238]:
data_set.tail(5)

City      Sport       Discipline       Event First_Name  \
Year                                                                 
2008.0  Beijing  Wrestling  Wrestling Gre-R   66 - 74kg       None   
2008.0  Beijing  Wrestling  Wrestling Gre-R   66 - 74kg       None   
2008.0  Beijing  Wrestling  Wrestling Gre-R   55 - 60kg       None   
2008.0  Beijing  Wrestling  Wrestling Gre-R   60 - 66kg       None   
2008.0  Beijing  Wrestling  Wrestling Gre-R  96 - 120kg       None   

          Last_Name Gender Country_Code     Country Event_gender   Medal  
Year                                                                      
2008.0   Christophe    Men          FRA      France            M  Bronze  
2008.0     Manuchar    Men          GEO     Georgia            M    Gold  
2008.0      Vitaliy    Men          AZE  Azerbaijan            M  Silver  
2008.0       Steeve    Men          FRA      France            M    Gold  
2008.0       Mijain    Men          CUB        Cuba            M    Gold

# Ejercicio 3

## Obtener medidas resumen del conjunto de datos. Cuál es el país que ganó más medallas?

In [239]:
data_set.describe()

City     Sport Discipline   Event First_Name  Last_Name Gender  \
count     15316     15316      15316   15316         14      15223  15316   
unique        9        28         41     293          2       5584      2   
top     Beijing  Aquatics  Athletics  hockey        Jr.   Vladimir    Men   
freq       2042      2210       1523     817         10        113   9388   

       Country_Code        Country Event_gender   Medal  
count         15316          15316        15316   15316  
unique          128            127            3       3  
top             USA  United States            M  Bronze  
freq           1992           1992         8817    5258

In [240]:
ds_by_country=data_set.groupby(by='Country')[['Medal']].count().sort_values(by='Medal',ascending=False)

In [241]:
ds_by_country.head(5)

Medal
Country             
United States   1992
Soviet Union    1021
Australia        798
Germany          691
China            679

### El país que ganó más medallas es Estados Unidos con un total de 1992 medallas.

### Cuantas medallas ganó Argentina?

In [242]:
ds_by_country.filter(items=['Argentina'],axis=0)

Medal
Country         
Argentina    153

# Ejercicio 4   
## pd: Agregué las medallas ganadas por mujeres con una tabla multiIndex para obtener más información

## Construir una tabla que muestre cuántas medallas obtuvieron los hombres en total en cada año que se realizó el evento.

In [343]:
data_set_by_year_gender=data_set.groupby(by=['Gender','Year'])[['Medal']].count()

In [344]:
data_set_by_year_gender['Change']=data_set_by_year_gender.diff()




In [345]:
data_set_by_year_gender.iloc[9,1]=0
data_set_by_year_gender=data_set_by_year_gender.fillna(value=0)

In [346]:
data_set_by_year_gender

Medal  Change
Gender Year                 
Men    1976.0    924     0.0
       1980.0    958    34.0
       1984.0    973    15.0
       1988.0   1011    38.0
       1992.0   1105    94.0
       1996.0   1082   -23.0
       2000.0   1126    44.0
       2004.0   1099   -27.0
       2008.0   1110    11.0
Women  1976.0    381     0.0
       1980.0    429    48.0
       1984.0    486    57.0
       1988.0    535    49.0
       1992.0    600    65.0
       1996.0    777   177.0
       2000.0    889   112.0
       2004.0    899    10.0
       2008.0    932    33.0

In [347]:
data_set_by_year_gender[['Medal Change %' , 'Change Difference %']]=data_set_by_year_gender.pct_change()

In [348]:
data_set_by_year_gender=data_set_by_year_gender.fillna(value=0)
data_set_by_year_gender[['Medal Change %' , 'Change Difference %']]=np.round(data_set_by_year_gender[['Medal Change %' , 'Change Difference %']] *100,decimals=2)


In [353]:
data_set_by_year_gender.iloc[9,1:4]=0
data_set_by_year_gender

Medal  Change  Medal Change %  Change Difference %
Gender Year                                                      
Men    1976.0    924     0.0            0.00                 0.00
       1980.0    958    34.0            3.68                  inf
       1984.0    973    15.0            1.57               -55.88
       1988.0   1011    38.0            3.91               153.33
       1992.0   1105    94.0            9.30               147.37
       1996.0   1082   -23.0           -2.08              -124.47
       2000.0   1126    44.0            4.07              -291.30
       2004.0   1099   -27.0           -2.40              -161.36
       2008.0   1110    11.0            1.00              -140.74
Women  1976.0    381     0.0            0.00                 0.00
       1980.0    429    48.0           12.60                  inf
       1984.0    486    57.0           13.29                18.75
       1988.0    535    49.0           10.08               -14.04
       1992.0    600    65.0           12.15                32.65
       1996.0    777   177.0           29.50               172.31
       2000.0    889   112.0           14.41               -36.72
       2004.0    899    10.0            1.12               -91.07
       2008.0    932    33.0            3.67               230.00

# Algunos detalles del punto 4.

### 0 - Si bien, en la mayoria de los deportes, los hombres y mujeres compiten por separado se nota que:
### 1- En terminos relativos el porcentaje de medallas que ganan las mujeres es mayor al de hombres. (¿Se suman mujeres a más deportes?)
### 2- En terminos absolutos, la disminución en la medallas ganadas por los hombres se debe a que: 
#### A- Hay más deportes compartidos y las mujeres estan ganando las medallas.
#### B- Se quitaron deportes de los juegos olimpicos.
### 3- El año que los hombres ganaron mas medallas (1126) fue en el 2000, El año que las mujeres ganaron mas medallas (932) fue en el 2008.

